internal/auth.go

```diff
@@ -119,9 +119,13 @@ func redirectBase(r *http.Request) string {
 
 // // Return url
 func returnUrl(r *http.Request) string {
+	if r.Header.Get("X-Replaced-Path") != "" {
```

Comment:
Please could you put r.Header.Get("X-Replaced-Path") into a variable rather than requesting it twice?

Reply:
Sure. done